lib/buildkite/test_collector/test_links_plugin/reporter.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Buildkite::TestCollector::TestLinksPlugin
+  class Reporter
+    RSpec::Core::Formatters.register self, :dump_failures
+
+    def initialize(output)
+      @output = output
+    end
+
+    def dump_failures(notification)
+      # Do not display summary if no failed examples
+      return unless notification.failed_examples.present?
+
+      # Check if a Test Analytics token is set
+      return unless Buildkite::TestCollector.api_token
+
+      metadata = fetch_metadata
+      
+      return if metadata.nil?
+
+      url = metadata.fetch('suite_url')
+
+      # If the suite_url does not exist, then we are unable to create the test links
+      return unless url
```

Comment:
Ok I had a bit of a think about this and I think this section should be:

```
metadata = fetch_metadata

# return if metadata was not fetched successfully
return if metadata.nil?

# return if metdata does not contain suite url
return if metadata["suite_url"].nil?
```

And then line 30 can become 
```
failed_example_output(example, metadata["suite_url"])
```
as there's no need to assign the suite_url to the url field anymore because it's not doing an outbound request (you can do the assignment if you wish but there isn't a functional reason to avoid it)

and lines 56-58 can go back to 
```
JSON.parse(response.body) if response.code == 200 
```

it would be worth writing tests for all scenarios of:
1. metadata wasn't requested
2. metadata was requested but there was an error thrown in the process
3. metadata was requested but server responds with non 200 response
4. metadata was requested but suite_url returned is nil
5. metadata was requested and suite_url is present

it would probably also be worth integration testing all these scenarios - give me a bell if you need a hand with that

Reply:
Perfect! I will give this ago. Also, I realised that it the response code is a string not boolean, that was why our original test failed